auth0/src/main/java/com/auth0/android/authentication/AuthenticationException.java

```diff
@@ -157,17 +157,22 @@ public boolean isInvalidConfiguration() {
 
     /// When MFA code is required to authenticate
     public boolean isMultifactorRequired() {
-        return "a0.mfa_required".equals(code);
+        return "mfa_required".equals(code) || "a0.mfa_required".equals(code);
     }
 
     /// When MFA is required and the user is not enrolled
     public boolean isMultifactorEnrollRequired() {
-        return "a0.mfa_registration_required".equals(code);
+        return "a0.mfa_registration_required".equals(code) || "unsupported_challenge_type".equals(code);
+    }
+
+    /// When the MFA Token received on the login request has expired
+    public boolean isMultifactorTokenExpired() {
```

Comment:
Why not the more generic `isMultifactorTokenInvalid` as an expired token is invalid, does it matter how? You also have `{"error":"invalid_grant","error_description":"Malformed mfa_token"}`

Reply:
I didn't catch that error response, probably because I tested it on Lock directly and the token is handled from one endpoint to the other as it is. Yeah it makes sense to handle both cases under the same method, and I agree that in that case it should be named like that -> Invalid.